### ISSUE TITLE
On motorways: Show the "Leave now"-arrow earlier.

### ISSUE THREAD

**Issue by Neuwessi11** [osmandapp/OsmAnd#12818]
### Description

Because I have no TTS installed, I cannot say anything about voice prompts, but when leaving a motorway the point of leaving the motorway (shown by the number of metres) is at the end of the turning lane.

### Describe the solution you'd like

In all cases if prior to a motorway exit there is tagged an extra lane (mostly in combination with turn:lane=slight_right or something), let the countdown of metres end at the beginning of the turning lane and not at it's end.
Because when driving you normally turn at the beginning of the turning lane. This will be very helpful not only on motorway exits, but even more on motorway junctions, because there the turning lanes are very much longer, and it feels always somewhat uncertain to leave the motorway when the app suggests me to leave 500m later.

### Describe alternatives you've considered

None.

**Comment by sonora** [comment]
Sounds inconsistent because also all road signage like the half mile/quarter mile warnings or the 300m/200m/100m signage on Europe's motorways more refer to a point where you actually fork from the motorway, not to the very beginning of turn lanes. If we created a different reference point in OsmAnd it may likely confuse more than help.

Having said that, chances are that a good chunk of your issue comes from not using TTS announcements. I recommend trying that, it may considerably change the navigation experience for you, and if not just for the fact that relying on visual only can constitute a safety risk. ;)

**Comment by Neuwessi11** [comment]
> Do you have the lane guidance widget turned on? Map data permitting, it should display exactly what you're asking for?

Yes, I have it turned on, and I like it very much in most cases. (By the way: Not in all cases. See #12618.)
And it refers always to the point highway=motorway_junction, what is (if correctly tagged) at the end of the exit lane.

> Sounds inconsistent because also all road signage like the half mile/quarter mile warnings or the 300m/200m/100m signage on Europe's motorways more refer to a point where you actually fork from the motorway, not to the very beginning of turn lanes. If we created a different reference point in OsmAnd it may likely confuse more than help.

Maybe it is indeed a little inconsistent.
But maybe sometimes inconsistence can be helpful ... :)

> Having said that, chances are that a good chunk of your issue comes from not using TTS announcements. I recommend trying that, it may considerably change the navigation experience for you, and if not just for the fact that relying on visual only can constitute a safety risk. ;)
> 
I didn't want to use Google's TTS, and I didn't find a good other one. (I really tried to find. If you have a good suggestion, I would be grateful ...)
And in my experience in every navigation system or navigation software the voice were the first thing to disable, because they in every case were so annoying.

**Comment by vshcherb** [comment]
This would be hard or even impossible to fix cause it involves also the mapping culture:  are there strict guidelines where mappers would indicate split, do they have precise information or they use vague satellite imagery, is it consistent in all regions. 

As of today OsmAnd represents information that's on map and it doesn't have information where "extra lanes" start and ends. Though even if that information is present, it is much more important to have consistent mapping & also consistent picture for all drivers what warning means and how it works.

**Comment by Neuwessi11** [comment]
> As of today OsmAnd represents information that's on map and it doesn't have information where "extra lanes" start and ends.

Not explizit, but implizit. 
If there is a lane prior to an exit or (in case of being in the middle of a junction) prior to a lane split that are tagged with slight_right or (in case of a junction) slight_left, this should be an "extra lane". Or am I wrong?
 
I know OsmAnd is not made for that, but just as a thougt experiment: If the software of an autonomous driven car tells the car not earlier than at the motorway_junction-node to "leave now", it would in many cases lead either to misnavigation or to a real mess, because there would be really little time for leaving the motorway. And it would quite often irritate all not-autonomous drivers, when leaving the motorway so late.
As I said: I know that autonomous driving is not your intention, but maybe it can make clearer what I mean.
